components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/endpoint/registry/util/GraphQLDefinitionValidator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.impl.endpoint.registry.util;
+
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import graphql.schema.idl.UnExecutableSchemaGenerator;
+import graphql.schema.idl.errors.SchemaProblem;
+import graphql.schema.validation.SchemaValidationError;
+import graphql.schema.validation.SchemaValidator;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Set;
+
+/**
+ * This class provides the functionality of validating GraphQL definitions
+ */
+public class GraphQLDefinitionValidator implements DefinitionValidator {
+    private static final Log log = LogFactory.getLog(GraphQLDefinitionValidator.class);
+
+    @Override
+    public boolean validate(URL definitionUrl) {
+        boolean isValid = false;
+        try {
+            String definitionContent = IOUtils.toString(definitionUrl.openStream());
+            isValid = validate(definitionContent);
+        } catch (IOException e) {
+            log.error("Error in reading content in the definition URL: " + definitionUrl, e);
+        }
```

Comment:
Rather than capturing and logging this, throw a ValidationException.

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8618/commits/169ace7202aa557c7b6710f5bf9e481af4ff99d4